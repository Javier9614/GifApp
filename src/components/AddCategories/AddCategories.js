import React, {useState} from 'react';
import PropTypes from 'prop-types';


const AddCategories = ({setCategories}) => {

    //si dejamos el useState() vacio nos saldra un warning
    //porque es undefined y digamos que eso a react no le gusta xD
    //si queremos dejarlo vacio lo ponemos como string vacio y fiesta
    const[inputValue, setInputValue] = useState("");


    const handleInputChange = (e) => {
        //Esto me permite cambiar el valor del input, es
        //porque de otra manera se quedaria estatico con 
        //el valor por defecto ("Categories")
        setInputValue(e.target.value);
        
    }


    const handleSubmit = (e) => {
        //Esto evita que se recargue la pagina al darle enter
        e.preventDefault();
        if(inputValue.trim().length>2){

            setCategories(categories => [inputValue,...categories]);
            setInputValue("");

        }
    }

    return(

        <form onSubmit={handleSubmit}>

            <input 
                type="text"
                value={inputValue}
                placeholder="Search all you need"
                onChange={handleInputChange}
            />
        </form>
    )
}
AddCategories.propTypes={
    setCategories: PropTypes.func.isRequired,
}
export default AddCategories;