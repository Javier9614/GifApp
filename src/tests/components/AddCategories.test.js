import React from 'react';
import { shallow } from "enzyme";
import AddCategories from "../../components/AddCategories/AddCategories";

describe('Pruebas en <AddCategories>',() => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategories setCategories={setCategories} />);

    beforeEach(() => {
        //esta funcion sirve pare reinicializar todo el testeo
        jest.clearAllMocks();
        wrapper = shallow(<AddCategories setCategories={setCategories} />);
        
    })

    test('Debe de mostrarse correctamente',() => {

        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de mostrar correctamente los cambios del input',() => {

        const input= wrapper.find("input");

        const value = "hola mundo";
        input.simulate('change', {target:{value:value }});
        
    })

    test('No debe de postear la info con submit',() => {

        wrapper.find('form').simulate('submit',{preventDefault(){}});

        expect( setCategories ).not.toHaveBeenCalled();
    })

    test("Debe llamar el setCategories y limpiar la caja de texto",()=>{

        
        const value = "hola mundo";
        wrapper.find("input").simulate('change', {target:{value}});

        wrapper.find("form").simulate('submit', {preventDefault(){}});

        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledTimes(1);
        expect( setCategories ).toHaveBeenCalledWith(expect.any(Function));

        expect(wrapper.find("input").prop("value")).toBe("");
        
    })

})