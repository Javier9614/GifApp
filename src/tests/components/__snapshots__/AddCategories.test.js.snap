// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategories> Debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Search all you need"
    type="text"
    value=""
  />
</form>
`;
